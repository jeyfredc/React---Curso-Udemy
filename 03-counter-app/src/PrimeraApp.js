
import './index.css'
import PropTypes from 'prop-types';

//Functional Components
const PrimeraApp = ({ saludo, subtitulo }) => {


    return (
        <>
        <h1>{ saludo } !!!</h1>
            {/* {}<pre>{ JSON.stringify(saludo, null, 3) }</pre> */}
            <p>{ subtitulo }</p>
        </>
    )
}

PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired
}

PrimeraApp.defaultProps = {
    subtitulo: 'Soy un subtitulo'
}

export default PrimeraApp;