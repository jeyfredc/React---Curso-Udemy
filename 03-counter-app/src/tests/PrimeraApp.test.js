import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import PrimeraApp from '../PrimeraApp';

describe('Pruebas en <PrimeraApp />', () => {

    test('Debe de mostrar el mensaje "Hola, Soy Iron Man"', () => {
        
        const saludo = 'Hola, Soy Iron Man';

        const { getByText} = render( <PrimeraApp saludo={ saludo } /> );

        expect( getByText(saludo)).toBeInTheDocument();
    })
    
    
})
