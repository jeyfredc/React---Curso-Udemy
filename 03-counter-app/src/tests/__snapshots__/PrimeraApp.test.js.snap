// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp /> debe de mostrar el componenet <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Hola, Soy Iron Man
     !!!
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
